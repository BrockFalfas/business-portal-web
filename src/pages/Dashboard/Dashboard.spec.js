import React from 'react';
import { shallow } from 'enzyme';

import { Dashboard } from './index';

const initDashboard = overrides => {
  const mockProps = {
    exchangeRates: { 'thor-usd': { rate: 1.0 } },
    syncTransactions: jest.fn(),
    syncWallet: jest.fn(),
    transactions: [],
    wallet: {},
  };
  const wrapper = shallow(<Dashboard {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('page: <Dashboard/>', () => {
  it('should render without crashing', () => {
    const { wrapper } = initDashboard();
    expect(wrapper).toBeTruthy();
  });

  it('should render as expected', () => {
    const { wrapper } = initDashboard({ wallet: { EUSD: 10, THOR: 20 } });
    expect(wrapper).toMatchSnapshot();
  });

  describe('lifecycle', () => {
    describe('componentDidMount', () => {
      const { mockProps, wrapper } = initDashboard();
      const instance = wrapper.instance();

      instance.componentDidMount();

      expect(mockProps.syncTransactions).toHaveBeenCalled();
      expect(mockProps.syncWallet).toHaveBeenCalled();
    });
  });

  describe('calcWalletBalance', () => {
    const { wrapper } = initDashboard();

    it('should calculate wallet balance', () => {
      wrapper.setProps({
        wallet: { EUSD: 100, THOR: 200 },
        exchangeRates: { 'thor-usd': { rate: 2 } },
      });

      const balance = wrapper.instance().calcWalletBalance();

      expect(balance).toEqual({
        usd: 100,
        thor: 200,
        thorAsUsd: 200 * 2,
        total: 100 + 400,
      });
    });

    it('should fallback exchange rate to 0 when not set', () => {
      wrapper.setProps({
        wallet: { EUSD: 100, THOR: 200 },
        exchangeRates: { 'thor-usd': {} },
      });

      const balance = wrapper.instance().calcWalletBalance();

      expect(balance).toEqual({
        usd: 100,
        thor: 200,
        thorAsUsd: 200 * 0,
        total: 100,
      });
    });

    it('should fallback usd to 0 when not set', () => {
      wrapper.setProps({
        wallet: { THOR: 200 },
        exchangeRates: { 'thor-usd': { rate: 2 } },
      });

      const balance = wrapper.instance().calcWalletBalance();

      expect(balance).toEqual({
        usd: 0,
        thor: 200,
        thorAsUsd: 200 * 2,
        total: 0 + 400,
      });
    });

    it('should fallback thor to 0 when not set', () => {
      wrapper.setProps({
        wallet: { EUSD: 100 },
        exchangeRates: { 'thor-usd': { rate: 2 } },
      });

      const balance = wrapper.instance().calcWalletBalance();

      expect(balance).toEqual({
        usd: 100,
        thor: 0,
        thorAsUsd: 0 * 2,
        total: 100 + 0,
      });
    });
  });

  describe('handleTransactionsPeriodChange', () => {
    it('should update transactionsPeriod state', () => {
      const { wrapper } = initDashboard();
      const instance = wrapper.instance();

      instance.handleTransactionsPeriodChange('MONTH');
      expect(wrapper.state('transactionsPeriod')).toBe('MONTH');
      instance.handleTransactionsPeriodChange('DAY');
      expect(wrapper.state('transactionsPeriod')).toBe('DAY');
    });
  });
});
