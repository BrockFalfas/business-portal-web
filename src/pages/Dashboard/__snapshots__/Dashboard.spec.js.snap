// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: <Dashboard/> should render as expected 1`] = `
<div
  className="dashboard"
>
  <Header
    size="medium"
    title="Account Summary"
  >
    <Actions
      position="top"
    >
      <Right>
        <Button
          ghost={true}
        >
          Manage Accounts
        </Button>
      </Right>
    </Actions>
  </Header>
  <Row
    gutter={32}
  >
    <Col
      lg={8}
      xs={24}
    >
      <Card
        color="black"
        description="USD Wallet Balance"
        icon={
          <Icon
            type="pie-chart"
          />
        }
        rounded={true}
        title="$10.00"
      >
        <Button
          ghost={true}
          size="small"
        >
          Transfer to Bank
        </Button>
      </Card>
    </Col>
    <Col
      lg={8}
      xs={24}
    >
      <Card
        color="blue"
        description="20 THOR"
        icon={
          <Icon
            type="pie-chart"
          />
        }
        rounded={true}
        title="$20.00"
      >
        <Button
          ghost={true}
          size="small"
        >
          BUY / SELL
        </Button>
      </Card>
    </Col>
    <Col
      lg={8}
      xs={24}
    >
      <Card
        color="green"
        description="Account Value"
        rounded={true}
        title="$30.00"
      />
    </Col>
  </Row>
  <Divider />
  <Header
    size="small"
    title="Recent Transactions"
  >
    <Actions
      position="top"
    >
      <Left>
        <Dropdown
          onClick={[Function]}
          options={
            Array [
              "DAY",
              "MONTH",
              "YEAR",
            ]
          }
        >
          <Button
            ghost={true}
            type="primary"
          >
            MONTH
             
            <Icon
              type="down"
            />
          </Button>
        </Dropdown>
         
        <Button
          type="primary"
        >
          Create Report
        </Button>
      </Left>
      <Right>
        <Link
          replace={false}
          to="#"
        >
          View All Transactions →
        </Link>
      </Right>
    </Actions>
  </Header>
  <Row
    gutter={32}
  >
    <Col
      lg={12}
    >
      <Box
        backgroundColor="white"
        className={null}
        padding={false}
        transparent={false}
      >
        <withTheme(ChartComponent)
          aggregatedValue="$135,067.89"
          component={[Function]}
          height="100"
          theme="green"
          title="Revenue"
        />
      </Box>
    </Col>
    <Col
      lg={12}
    >
      <Box
        backgroundColor="white"
        className={null}
        padding={false}
        transparent={false}
      >
        <withTheme(ChartComponent)
          aggregatedValue="$90,874.54"
          component={[Function]}
          height="100"
          theme="blue"
          title="Transactions"
        />
      </Box>
    </Col>
  </Row>
  <Divider />
  <Header
    size="small"
    title="Satisfaction Rates"
  >
    <Actions
      position="top"
    >
      <Left>
        <Button
          type="primary"
        >
          Conduct Survey
        </Button>
      </Left>
      <Right>
        <Link
          replace={false}
          to="#"
        >
          View All Surveys →
        </Link>
      </Right>
    </Actions>
  </Header>
  <div>
    <Row
      gutter={32}
    >
      <Col
        lg={8}
        xs={24}
      >
        <Box
          backgroundColor="white"
          className={null}
          padding={false}
          transparent={false}
        >
          <SurveySlider
            description="How satisfied are you with our new product?"
            max={100}
            min={0}
            onAfterChange={[Function]}
            votesCount={25}
          />
        </Box>
      </Col>
      <Col
        lg={8}
        xs={24}
      >
        <Box
          backgroundColor="white"
          className={null}
          padding={false}
          transparent={false}
        >
          <SurveySlider
            description="How satisfied are you with your workload?"
            max={100}
            min={0}
            onAfterChange={[Function]}
            votesCount={31}
          />
        </Box>
      </Col>
      <Col
        lg={8}
        xs={24}
      >
        <Box
          backgroundColor="white"
          className={null}
          padding={false}
          transparent={false}
        >
          <SurveySlider
            description="Would you recommend working for Luber to a friend?"
            max={100}
            min={0}
            onAfterChange={[Function]}
            votesCount={46}
          />
        </Box>
      </Col>
    </Row>
  </div>
</div>
`;
